src/bindings/cpp/examples/cpp_example_hierarchy.cpp

```diff
@@ -22,8 +22,8 @@ namespace kdb
  * @param k the key to search in
  * @param n the level
  *
- * nth_level_of_name(Key("user/hello", KEY_END), 0) -> "user"
- * nth_level_of_name(Key("user/hello", KEY_END), 1) -> "hello"
+ * nth_level_of_name(Key("user:/hello", KEY_END), 0) -> "user:"// TODO (kodebach): ?????
```

Comment:
@markus2330 I am not sure what this example is supposed to show, but with the new keyname structure, this function is very weird

Reply:
Why?

Reply:
Well the whole example is weird. First of all it doesn't use unescaped names and only works with `user:/` and `system:/` keys. Then it builds an expensive `KeyHierarachy`, but that doesn't have a lot more (if any more) functionality than a standard `KeySet`. Its also C++ code so it should really use the `NameIterator` instead of this function.

However, what I am concerned about here, is that `user:` should not be seen as a level of the keyname. That's the whole point of the `:`. The stuff before the `:/` is the namespace. The stuff after is the actual name.

Anyways, since this is just an example, it doesn't really matter what it does, so I'll just leave it like this...

Reply:
Yes you are right: nth_level_of_name is also totally buggy, it would need a rewrite with the name iterator. So the example can be removed.